QuickSortAlgorithm.java

```diff
@@ -0,0 +1,99 @@
+/**
+ * Recursive function to sort the given array in ascending order using quick
+ * sort Algorithm.
+ * 
+ * @author Udit Saxena
+ *
+ */
+
+public class QuickSortAlgorithm {
+	/**
+	 * function for finding partition Index
+	 * 
+	 * @param array
+	 *            - integer array (Unsorted)
+	 * @param startIndex
+	 *            - starting index of array
+	 * @param endIndex
+	 *            - end index of array
+	 * @return partitionIndex 
+	 * 			  -index from where array is divided into two part
+	 */
+	int returnPartitionIndex(int[] array, int startIndex, int endIndex) {
+		try {
+			checkEmptyArray(array);
+			int partitionIndex = startIndex - 1;
+			int pivot = array[endIndex];
+			for (int i = startIndex; i < endIndex; i++) {
+				if (array[i] < pivot) {
+					partitionIndex++;
+					swap(array, i, partitionIndex);
+				}
+			}
+			swap(array, endIndex, partitionIndex + 1);
+			return partitionIndex + 2;
+		} catch (Exception e) {
+			System.out.println(e);
+			return 0;
+		}
+	}
+
+	/**
+	 * recursive function of quick sort algorithm
+	 * 
+	 * @param array
+	 *            - integer array (Unsorted)
+	 * @param startIndex
+	 *            - starting index of array
+	 * @param endIndex
+	 *            - end index of array
+	 */
+	void quickSort(int[] array, int startIndex, int endIndex) {
+		try {
+			checkEmptyArray(array);
```

Comment:
Not required, as start index and end index would be equal for empty array and below sorting logic will be skipped